### ISSUE TITLE
High Fee Mitigation: Trusted Nodes

### ISSUE THREAD

**Issue by reemuru** [monero-project/monero-gui#3898]
> Update remote node scanner from simple mode and use hardcoded trusted community nodes.

@selsta 

Reference:

#3897

**Comment by jeffro256** [comment]
If a PR is to be created from this, make sure to also use hardcoded SSL certificates

**Comment by jeffro256** [comment]
I think we should keep the public node list, but each entry must be verified by a chain of SSL certificates which is signed by one of a small set of hard coded root certificates, much like an HTTPS-like certificate authority.

**Comment by selsta** [comment]
But there is no "public node list". Currently the daemon is used for a p2p node scanner, but I want to remove that but it removes the dependency on the daemon. I'll look into SSL.

**Comment by jeffro256** [comment]
We could create an RPC command that retrieves a daemon's p2p public node list. So on start, the wallet runs the command `get_public_nodes` (or similar) on a *trusted* node, caches the list, randomly picks one from the list, connects to it, checks that it has a verified SSL certificate, and only then starts syncing and asking it for fee estimations. This would allow for a safer environment for users while still remaining scalable and decentralized (not all traffic routed through a small set of nodes). 

Furthermore, the node information (IP + SSL cert) could be saved for each transaction in the wallet file. In the case that a node gave a wallet a ridiculous fee / bad decoys / etc, one could pull up the SSL signed message and "blame" that cert. They could copy/paste that information to the devs and further steps could be taken to revoke that certificate.

**Comment by selsta** [comment]
There are two things to consider:

- Passive spy nodes (setup by blockchain analysis companies)
- Actively malicious nodes

I would like to avoid both. Yes, it's possible for trusted community nodes to spy too, but the chance is way lower than when selecting a random node in the network.

I'll check into SSL certs, but currently I only plan to use a trusted node list.

**Comment by jeffro256** [comment]
> Yes, it's possible for trusted community nodes to spy too, but the chance is way lower than when selecting a random node in the network.

Yeah I'm saying that it shouldn't broadcast random nodes in the network, only those who are signed by a small group of hardcoded community certs

> I'll check into SSL certs, but currently I only plan to use a trusted node list.

This will still maintain the "trusted" aspect of your idea, but now the trusted nodes don't have to be tied down to the same IP address, they can add server nodes and delegate load to them. Also, if someone from the trusted group loses their public IP (changing hosting sites for example) then they can easily move their server without everyone having to recompile their code / download new binaries.
